src/utility.ts

```diff
@@ -376,14 +360,10 @@ export class Utility {
         return set;
     }
 
-    private static async getEdgeDeviceList(iotHubConnectionString: string): Promise<any[]> {
-        const body = {
-            query: "SELECT * FROM DEVICES where capabilities.iotEdge=true",
-        };
-        const url = `/devices/query?api-version=${Constants.IoTHubApiVersion}`;
-        const config = Utility.generateIoTHubAxiosRequestConfig(iotHubConnectionString, url, "post", body);
-
-        return (await axios.request(config)).data;
+    private static async getEdgeDeviceList(iotHubConnectionString: string): Promise<Twin[]> {
+        const registry: Registry = Registry.fromConnectionString(iotHubConnectionString);
+        const query = registry.createQuery("SELECT * FROM DEVICES where capabilities.iotEdge=true");
+        return ((await query.nextAsTwin(null)) as ResultWithIncomingMessage<Twin[]>).result;

```

Comment:
why the return need to be twin?

Reply:
Just for better coding experience. The return type could be `Twin` or `any`. Use `Twin`, then we could have  IntelliSense.